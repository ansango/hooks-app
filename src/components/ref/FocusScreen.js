import { useRef } from "react";

export const FocusScreen = () => {
  const inputRef = useRef();
  const handleClick = () => {
    inputRef.current.select();
  };

  return (
    <>
      <input ref={inputRef} className="form-control" placeholder="Your name" />
      <button className="btn btn-outline-primary mt-5" onClick={handleClick}>
        Focus
      </button>
    </>
  );
};
