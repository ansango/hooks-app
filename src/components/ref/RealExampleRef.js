import { useState } from "react";
import { MultipleCustomHooks } from "../custom/MultipleCustomHooks";

export const RealExampleRef = () => {
  const [show, setShow] = useState(false);
  return (
    <>
      <button
        className="btn btn-primary"
        onClick={() => {
          setShow(!show);
        }}
      >
        Show/Hide
      </button>

      {show && <MultipleCustomHooks />}
    </>
  );
};
