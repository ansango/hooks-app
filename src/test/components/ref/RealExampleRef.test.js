import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/ref/RealExampleRef";

describe("Test in <RealExampleRef/>", () => {
  const wrapper = shallow(<RealExampleRef />);
  test("should render", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).not.toBe(true);
  });

  test("should show <MultipleCustomHooks/>", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
