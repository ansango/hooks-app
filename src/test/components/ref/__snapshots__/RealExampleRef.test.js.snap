// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <RealExampleRef/> should render 1`] = `
<Fragment>
  <button
    className="btn btn-primary mb-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
